### PR TITLE
improve e2e tests for supporting mac env and coredns autonomy

### PR DESCRIPTION
**Pull request by rambohe-ch**
/kind enhancement

#### What this PR does / why we need it:
improve e2e tests as following:
1. add a parameter `ENABLE_AUTONOMY_TESTS` for enabling or disabling autonomy e2e tests, default is: true(enable autonomy e2e tests)
2. add a parameter `TARGET_PLATFORMS` for specifying platform that run e2e tests. for example: the command for running e2e tests on mac env: make e2e-tests TARGET_PLATFORMS=linux/arm64
3. enable autonomy tests for CoreDNS component.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

**Comment by rambohe-ch** [comment]
@Congrool @lorrielau PTAL

**Comment by Congrool** [comment]
It seems that 10s timeout is too short for recover coredns. We can change it to 30s.

**Comment by rambohe-ch** [comment]
> It seems that 10s timeout is too short for recovering coredns. We can change it to 30s.

@Congrool Thanks for your feedback, i have revised the timeout duration to 30s.

File: test/e2e/autonomy/autonomy_test.go [hunk 1]
```diff
@@ -42,10 +42,10 @@ const (
 	YurtDefaultNamespaceName = "default"
 	YurtSystemNamespaceName  = "kube-system"
 	YurtCloudNodeName        = "openyurt-e2e-test-control-plane"
-	YurtEdgeNodeName         = "openyurt-e2e-test-worker"
-	YurtEdgeNode2Name        = "openyurt-e2e-test-worker2"
-	NginxServiceName         = "yurt-e2e-test-nginx"
-	CoreDNSServiceName       = "kube-dns"
+	//YurtEdgeNodeName         = "openyurt-e2e-test-worker"
+	//YurtEdgeNode2Name        = "openyurt-e2e-test-worker2"
```
Comment (Congrool) [inline test/e2e/autonomy/autonomy_test.go:46]: please remove this 2 lines

File: test/e2e/autonomy/autonomy_test.go [hunk 2]
```diff
@@ -181,27 +180,27 @@ var _ = ginkgo.Describe("edge-autonomy"+YurtE2ENamespaceName, ginkgo.Ordered, gi
 
 	var _ = ginkgo.Describe("coredns"+YurtE2ENamespaceName, func() {
 		ginkgo.It("coredns edge-autonomy test", ginkgo.Label("edge-autonomy"), func() {
-			ginkgo.Skip("current coredns does not support edge-autonomy, coredns-edge-autonomy tests will be skipped.")
-			// obtain coredns containerID with crictl
-			cmd := ` /bin/bash -c "crictl ps | grep coredns | awk '{print \$1}'"`
-			opBytes, err := exec.Command("/bin/bash", "-c", "docker exec -t "+coreDnsNodeName+cmd).CombinedOutput()
+			//ginkgo.Skip("current coredns does not support edge-autonomy, coredns-edge-autonomy tests will be skipped.")
```
Comment (Congrool) [inline test/e2e/autonomy/autonomy_test.go:183]: please remove this line.

File: test/e2e/autonomy/autonomy_test.go (hunk 1 above)
Comment (rambohe-ch) [reply test/e2e/autonomy/autonomy_test.go]: fixed

File: test/e2e/autonomy/autonomy_test.go (hunk 2 above)
Comment (rambohe-ch) [reply test/e2e/autonomy/autonomy_test.go]: fixed

**Review by Congrool — COMMENTED** [review]
/lgtm

**Comment by rambohe-ch** [comment]
Fixes #1037

**Pull request merged**